### ISSUE TITLE
[BUG] Export of Open Images dataset results in "labels empty in ...\lables\train.cache

### ISSUE THREAD

**Issue by matthewDDennis** [voxel51/fiftyone#3342]
### System information

-   **OS Platform and Distribution**  Windows 11
-   **Python version** (`3.9`):
-   **FiftyOne version** (`0.21.4`):
-   **FiftyOne installed from** (pip):

### Describe the problem

I create a YoloV5 dataset by
 - Creating a main Dataset
 - for each of the required classes I 
    - create a class dataset to load the information for the class
    - load a required number of images for [train, test, validation] 
    - and merge the class dataset with the main dataset
 - Export the main dataset in YoloV5 format

Sometimes, when I use the YoloV5 dataset to train I get an error that "all labels empty in XXX\datasets\\vehicle3\\labels\\train.cache, can not start training. See https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data".

Checking the label .txt files, they are all empty, so the YoloV5 training fails.

This morning it ran once successfully getting 100 images for[ Cat, Dog] Then I tried with [Car, Bus] and I got the error.  After that I always get the error.

### Code to reproduce issue

This is the method
~~~python
    def create_dataset(self, model_name, classes, num_images):
        """ Downloads a dataset """

        splits = ['train', 'validation', 'test']
        label_types = ["detections"]

        if fo.dataset_exists(model_name):
            fo.delete_dataset(model_name)

        try:
            dataset = fo.Dataset(model_name)
        except Exception as e:
            # Somehow it's possible to create a Schrodinger's Dataset: it doesn't
            # exist but can't be created. Let's delete it and create it. If it
            # throws again, it will percolate up.
            fo.delete_dataset(model_name)
            dataset = fo.Dataset(model_name)

        available_classes = fouo.get_classes()

        for thing in classes:
            
            if self.cancel_requested:
                return
            
            thing = thing.capitalize()

            if not thing in available_classes:
               self.log(LogMethod.Info|LogMethod.Server, {
                   "message": f"There is no class '{thing}' in the Open Images dataset. Skipping.",
                   "loglevel": "warning"
               })
               continue

            thing_desc = thing if thing.endswith('s') else f"{thing}s"

            self.training_step = f"Downloading {thing_desc} from Open Images"

            self.log(LogMethod.Info|LogMethod.Server, {
                "message": f"Downloading {thing_desc} from Open Images",
                "loglevel": "information"
            })

            subset_name = f'{model_name}-{thing}'
            if fo.dataset_exists(subset_name):
                fo.delete_dataset(subset_name)
        
            if self.cancel_requested:
                return

            try:
                class_dataset = foz.load_zoo_dataset('open-images-v7',
                                                     #splits=splits, # Don't specify a split, so we get all of them
                                                     label_types=label_types,
                                                     classes = thing,
                                                     max_samples=num_images,
                                                     seed=42,
                                                     shuffle=True,
                                                     dataset_name=subset_name)
            except Exception as e:
                self.report_error(e, __file__, str(e))
                class_dataset = None

            self.dataset_classes_processed += 1

            if self.cancel_requested:
                return

            if class_dataset and len(class_dataset):
                # Merge the sub_dataset into the main dataset
                dataset.merge_samples(class_dataset) 
                
                self.log(LogMethod.Info|LogMethod.Server, {
                    "message": f"Downloaded {len(class_dataset)} images of {thing_desc} from Open Images",
                    "loglevel": "information"
                })
            else:
                self.log(LogMethod.Info|LogMethod.Server, {
                    "message": f"No images of class {thing} were found in Open Images",
                    "loglevel": "warning"
                })
                
        if self.cancel_requested:
            return

        # Export the Dataset
        export_dir = f'{self.datasets_dirname}/{model_name}'

        self.training_step = f"Preparing Export of Dataset to '{export_dir}'"
        self.log(LogMethod.Info|LogMethod.Server, {
            "message": f"Preparing Export of Dataset to '{export_dir}'",
            "loglevel": "information"
        })

        if os.path.exists(export_dir):
            shutil.rmtree(export_dir)

        # retag the validation split. Careful with the "val" / "validation"!
        view = dataset.match_tags("validation")
        view.tag_samples("val")
        view.untag_samples("validation")

        # Export the splits
        splits = ['train', 'val', 'test']

        for split in splits:
            if self.cancel_requested:
                return

            self.training_step = f"Export {split} set to '{export_dir}'"
            split_view = dataset.match_tags(split)
            split_view.export(export_dir=export_dir,
                              dataset_type=fo.types.YOLOv5Dataset,
                              split=split,
                              classes=classes)
~~

### What areas of FiftyOne does this bug affect?

-   [ X] `App`: FiftyOne application issue
-   [ ] `Core`: Core Python library issue
-   [ ] `Server`: FiftyOne server issue

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [ X] Yes. I would be willing to contribute a fix for this bug with guidance from the FiftyOne community
-   [ ] No. I cannot contribute a bug fix at this time
